scripts/rspec/rspec.ps1

```diff
@@ -243,44 +243,41 @@ function RemoveText($textToRemove, $modifiableString)
 }
 
 function AppendVbTestCase($ruleTestsFolder) {
-    $existingClassText = Get-Content -Path "${ruleTestsFolder}\\${csClassName}Test.cs" -Raw
-    $snippetText = Get-Content -Path "${RuleTemplateFolder}\\VbNetTestSnippet.cs" -Raw
-
     $usingToken = "using SonarAnalyzer.Rules.CSharp;"
-    $csUsingToken = "using CS = SonarAnalyzer.Rules.CSharp;"
-    $vbNetUsingToken = "using VB = SonarAnalyzer.Rules.VisualBasic;"
+    $usingTokenCS = "using CS = SonarAnalyzer.Rules.CSharp;"
+    $usingTokenVB = "using VB = SonarAnalyzer.Rules.VisualBasic;"
     $namespaceToken = "namespace SonarAnalyzer.UnitTest.Rules"
     $token = "    }"
-    $usingTokenIdx = $existingClassText.IndexOf($usingToken)
 
-    $newText = $existingClassText
+    $text = Get-Content -Path "${ruleTestsFolder}\\${csClassName}Test.cs" -Raw
+    $methodVB = Get-Content -Path "${RuleTemplateFolder}\\TestMethod.VB.cs" -Raw
+    $methodVB = $methodVB.Replace("<", "<VB.")
+
+    $usingTokenIdx = $text.IndexOf($usingToken)
     if ($usingTokenIdx -gt -1) {
-        $newText = $existingClassText.Remove($usingTokenIdx, $usingToken.Length + 1)
+        $text = $text.Remove($usingTokenIdx, $usingToken.Length + 1)
     }
 
-    $idx = $newText.LastIndexOf($token)
-
+    $idx = $text.LastIndexOf($token)
     if ($idx -gt -1) {
-        $newText = $newText.Remove($idx, $token.Length).Insert($idx, "`r`n${snippetText}`r`n${token}")
```

Comment:
You most probably did but I have to ask, did you try to run the case when the VbNet test case is appended. I see that you changed the logic here, and if I am not mistaken it was there because there was some problem with the new lines when appending the snippet.

Reply:
Yes I did. 
After changing the CR LF to LF, the problem with extra EOL reappeared and this was actually my fix for that. And that's how I found that this line was weird.